rest/config.go

```diff
@@ -331,7 +334,140 @@ func (dbConfig *DbConfig) setup(name string) error {
+		if err != nil {
+			return &JavaScriptLoadError{
+				JSLoadType: ImportFilter,
+				Path:       *dbConfig.ImportFilter,
+				Err:        err,
+			}
+		}
+		dbConfig.ImportFilter = &importFilter
+	}
+
+	// Load Conflict Resolution Function.
+	for _, rc := range dbConfig.Replications {
+		if rc.ConflictResolutionFn != "" {
+			conflictResolutionFn, err := loadJavaScript(rc.ConflictResolutionFn)
+			if err != nil {
+				return &JavaScriptLoadError{
+					JSLoadType: ConflictResolver,
+					Path:       rc.ConflictResolutionFn,
+					Err:        err,
+				}
+			}
+			rc.ConflictResolutionFn = conflictResolutionFn
+		}
+	}
+
+	return nil
+}
+
+// loadJavaScript loads the JavaScript source from an external file or and HTTP/HTTPS endpoint.
+// If the specified path does not qualify for a valid file or an URI, it returns the input path
+// as-is with the assumption that it is an inline JavaScript source. Returns error if there is
+// any failure in reading the JavaScript file or URI.
+func loadJavaScript(path string) (js string, err error) {
+	rc, err := readFromPath(path)
+	switch err.(type) {
+	case nil:
+		defer func() { _ = rc.Close() }()
+		src, err := ioutil.ReadAll(rc)
+		if err != nil {
+			return "", err
+		}
+		return string(src), nil
+	case *PathNotFoundError:
+		// If rc is nil and readFromPath returns no error, treat the
+		// the given path as an inline JavaScript and return it as-is.
+		return path, nil
+	default:
+		return "", err
+	}
+}
+
+// JSLoadType represents a specific JavaScript load type.
+// It is used to uniquely identify any potential errors during JavaScript load.
+type JSLoadType int
+
+const (
+	SyncFunction     JSLoadType = iota // Sync Function JavaScript load.
+	ImportFilter                       // Import filter JavaScript load.
+	ConflictResolver                   // Conflict Resolver JavaScript load.
+)
+
+// String returns the string representation of a specific JSLoadType.
+func (t JSLoadType) String() string {
+	jsLoadTypes := [...]string{"SyncFunction", "ImportFilter", "ConflictResolver"}
+	if len(jsLoadTypes) < int(t) {
+		return ""
+	}
+	return jsLoadTypes[t]
+}
+
+// JavaScriptLoadError is returned if there is any failure in loading JavaScript
+// source from an external file or URL (HTTP/HTTPS endpoint).
+type JavaScriptLoadError struct {
+	JSLoadType JSLoadType // A specific JavaScript load type.
+	Path       string     // Path of the JavaScript source.
+	Err        error      // Underlying error.
+}
+
+// Error returns string representation of the JavaScriptLoadError.
+func (e *JavaScriptLoadError) Error() string {
+	return fmt.Sprintf("Error loading JavaScript (%s) from (%q), Err: %v", e.JSLoadType, e.Path, e.Err)
+}
+
+// PathNotFoundError is returned when the specified path or URL (HTTP/HTTPS endpoint)
+// doesn't exist while constructing the ReadCloser.
+type PathNotFoundError struct {
+	Path string // Specified path or URL (HTTP/HTTPS endpoint).
+}
```

Comment:
Sure, PathNotFoundError has been replaced with ErrPathNotFound. The loadJavaScript doesn't really need to return more verbose message that includes the value of the path here because, we just need to read the path as-is with the assumption that it is an inline JavaScript in that case.